src/aws-healthomics-mcp-server/awslabs/aws_healthomics_mcp_server/tools/data_import_tools.py

```diff
@@ -0,0 +1,470 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data import tools for the AWS HealthOmics MCP server."""
+
+import botocore.exceptions
+import os
+import re
+from awslabs.aws_healthomics_mcp_server.utils.aws_utils import (
+    create_aws_client,
+)
+from loguru import logger
+from mcp.server.fastmcp import Context
+from pydantic import Field
+from typing import Any, Dict, List, Optional
+from urllib.parse import urlparse
+
+
+def parse_s3_uri(s3_uri: str) -> Dict[str, str]:
```

Comment:
As @markjschreiber mentioned earlier, there are some similarities with #1501, which was merged shortly after this was published.

Several of these utility functions now exist in `main` with equivalent functionality. I’d recommend doing another quick pass to see if any can be consolidated. For instance, `parse_s3_uri` could be replaced or simplified by adjusting the output of the existing [s3_utils.py](https://github.com/awslabs/mcp/blob/main/src/aws-healthomics-mcp-server/awslabs/aws_healthomics_mcp_server/utils/s3_utils.py#L44) implementation.